homeassistant/components/esphome/repairs.py

```diff
@@ -2,11 +2,108 @@
 
 from __future__ import annotations
 
+from typing import cast
+
+import voluptuous as vol
+
+from homeassistant import data_entry_flow
 from homeassistant.components.assist_pipeline.repair_flows import (
     AssistInProgressDeprecatedRepairFlow,
 )
 from homeassistant.components.repairs import RepairsFlow
-from homeassistant.core import HomeAssistant
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import issue_registry as ir
+
+from .manager import async_replace_device
+
+
+class ESPHomeRepair(RepairsFlow):
+    """Handler for an issue fixing flow."""
+
+    def __init__(self, data: dict[str, str | int | float | None] | None) -> None:
+        """Initialize."""
+        self._data = data
+        super().__init__()
+
+    @callback
+    def _async_get_placeholders(self) -> dict[str, str]:
+        issue_registry = ir.async_get(self.hass)
+        issue = issue_registry.async_get_issue(self.handler, self.issue_id)
+        assert issue is not None
+        return issue.translation_placeholders or {}
+
+
+class DeviceConflictRepair(ESPHomeRepair):
+    """Handler for an issue fixing device conflict."""
+
+    @property
+    def entry_id(self) -> str:
+        """Return the config entry id."""
+        assert isinstance(self._data, dict)
+        return cast(str, self._data["entry_id"])
+
+    @property
+    def mac(self) -> str:
+        """Return the MAC address of the new device."""
+        assert isinstance(self._data, dict)
+        return cast(str, self._data["mac"])
+
+    async def async_step_init(
+        self, user_input: dict[str, str] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the first step of a fix flow."""
+        return await self.async_step_start()
+
+    async def async_step_start(
+        self, user_input: dict[str, str] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the start step of a fix flow."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="start",
+                data_schema=vol.Schema(
+                    {
+                        vol.Required("action"): vol.In(
+                            {
+                                "migrate": "Migrate to new hardware",
+                                "manual": "Remove or rename the device",
+                            }
```

Comment:
Ideally these would be translatable, but I don't think fix_flows support that yet AFAICT

Reply:
Maybe I can use `menu_options` instead